src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/WindowStatement.scala

```diff
@@ -26,11 +31,70 @@ private[querygeneration] object WindowStatement {
     expAttr: (Expression, Seq[Attribute])
   ): Option[SnowflakeSQLStatement] = {
     val expr = expAttr._1
+    val fields = expAttr._2
 
     Option(expr match {
-      case _: RowNumber => ConstantString(expr.prettyName.toUpperCase) + "()"
+      // Handle Window Expression. (It is moved from MiscStatement)
+      case WindowExpression(func, spec) =>
+        func match {
+          // These functions in Snowflake support a window frame.
+          // Note that pushdown for these may or may not yet be supported in the connector.
+          case _: Rank | _: DenseRank | _: PercentRank =>
+            convertStatement(func, fields) + " OVER " + windowBlock(
+              spec,
+              fields,
+              useWindowFrame = true
+            )
+
+          // These do not.
+          case _ =>
+            convertStatement(func, fields) + " OVER " + windowBlock(
+              spec,
+              fields,
+              useWindowFrame = false
+            )
+        }
+
+      // Handle supported window function
+      case _: RowNumber | _: Rank | _: DenseRank =>
+        ConstantString(expr.prettyName.toUpperCase) + "()"
+
+      // Until July 21 2020, PercentRank can't be pushdown to snowflake.
```

Comment:
This is a bad comment, remove dates etc.

Reply:
update it.